### ISSUE TITLE
X11: Cursor in captured mode is broken for devices working in absolute mode

### ISSUE THREAD

**Issue by capnm** [godotengine/godot#21751]
**Godot version:**

f50dd682f6a0f9506daaf16fed06f4bc9efa6599

**OS/device including version:**

Linux / Ubuntu 18.04.1 LTS

**Issue description:** **Steps to reproduce:**

Set e.g. Wacom pen in absolute mode with the Gnome settings-panel or:
```
xsetwacom --set Wacom Intuos4 4x6 Pen stylus Mode Absolute
xsetwacom --set Wacom Intuos4 4x6 Pen stylus Area 0 0 31496 9842
xsetwacom --set Wacom Intuos4 4x6 Pen stylus MapToOutput desktop
```
The cursor leaves our planet:
![peek 2018-09-04 13-27](https://user-images.githubusercontent.com/4047289/45028744-7f9d1380-b046-11e8-804b-456adaf1be89.gif)

**Minimal reproduction project:**

Modified example from  https://github.com/godotengine/godot/issues/21744 (thanks!): 
[i21744-Constrained-Cursor.zip](https://github.com/godotengine/godot/files/2348401/i21744-Constrained-Cursor.zip)

**Comment by guilhermefelipecgs** [comment]
I didn't test it but maybe #19191 will fix this issue too. Like I said here https://github.com/godotengine/godot/issues/21744#issuecomment-418393418, capture mode on linux is very hack.

**Comment by capnm** [comment]
Yeah, the source code is already full of frustration ;)
Unfortunately it breaks the device with absolute coordinates even more :-/

![peek 2018-09-05 08-41](https://user-images.githubusercontent.com/4047289/45075973-16baa780-b0e9-11e8-8cee-f6c042f74724.gif)
 
[i21744-Constrained-Cursor-v3.zip](https://github.com/godotengine/godot/files/2351548/i21744-Constrained-Cursor-v3.zip)
[0001-Use-XInput2-RawMotion-to-generate-MouseMotion-events.patch.txt](https://github.com/godotengine/godot/files/2351571/0001-Use-XInput2-RawMotion-to-generate-MouseMotion-events.patch.txt)

**Comment by capnm** [comment]
> The cursor doesn't leaves our planet anymore.

Almost. Ok, let's add one circle ;-)

![peek 2018-09-05 21-01](https://user-images.githubusercontent.com/4047289/45115374-f540d680-b14f-11e8-9b6e-4c3bc5d07e31.gif)

[i21744-Constrained-Cursor-v5.zip](https://github.com/godotengine/godot/files/2354229/i21744-Constrained-Cursor-v5.zip)

**Comment by guilhermefelipecgs** [comment]
This is not a bug, in absolute mode this is expected because cursor will always forced to stay exact where the pen is, because of that, if your pen is not in center of the wacom pad the relative motion will be updated ever time.

Comparasion:
Mouse:
![peek 05-09-2018 16-43](https://user-images.githubusercontent.com/1387165/45117054-fb708c00-b12a-11e8-8213-2dadbba03739.gif)
Pen with absolute:
![peek 05-09-2018 16-50](https://user-images.githubusercontent.com/1387165/45117379-d16b9980-b12b-11e8-876c-56898ee5a87a.gif)

The only difference is when mouse stop moving, the relative motion (white dot) stop, but pen with absolute mode this is not possible. Only way to get the same result is changing the wacom to "Relative" mode.

**Comment by guilhermefelipecgs** [comment]
> Isn't it possible to calculate it from the incoming data?

Maybe it's possible. I think the only way is faking relative mode when using absolute mode on tablet but I don't know how to check that info.

**Comment by guilhermefelipecgs** [comment]
Also, I don't know if it's worth implementing.

**Comment by guilhermefelipecgs** [comment]
Games in general or made with godot?

**Comment by capnm** [comment]
Anything that does not work with absolute tracking mode, including Godot. We use VMware to manage the computers in the schoolroom ...

**Comment by guilhermefelipecgs** [comment]
Because blender don't use captured mode for free camera, it's use hidden.
![peek 06-09-2018 13-06](https://user-images.githubusercontent.com/1387165/45170211-b9e8eb00-b1d5-11e8-9eb8-320502d9e9c3.gif)

**Comment by guilhermefelipecgs** [comment]
I notice now that "free cam" works like blender fine in godot 3.0.6. We have a regression here.

**Comment by guilhermefelipecgs** [comment]
I'll open an issue about that.

**Comment by guilhermefelipecgs** [comment]
> I notice now that "free cam" works similar to blender in godot 3.0.6. We have a regression here.

Oops, forget that. I'm sleepy and winding things up here. :)

**Comment by capnm** [comment]
This issue is still not resolved in absolute tracking mode
 https://github.com/godotengine/godot/issues/19692

**Comment by cosmicchipsocket** [comment]
Should be fixed in #19191 now.

**Comment by capnm** [comment]
With the two PRs applied to the master, everything (free-cam, editor, ...) seems to work!

[0001-Fixes-to-mouse-mode-confined-and-captured.patch.txt](https://github.com/godotengine/godot/files/2366405/0001-Fixes-to-mouse-mode-confined-and-captured.patch.txt)
[0002-Use-XInput2-RawMotion-to-generate-MouseMotion-events.patch.txt](https://github.com/godotengine/godot/files/2366407/0002-Use-XInput2-RawMotion-to-generate-MouseMotion-events.patch.txt)


The only thing that catches my eye at the moment, is that the movement to the tablet edge (window-centered cursor) does not reach the edge of the window. (I've mapped the tablet surface to the entire desktop):
![screenshot1](https://user-images.githubusercontent.com/4047289/45295065-bae38a80-b4fd-11e8-9a50-c9dda8fbe7c4.jpg)

![screenshot2](https://user-images.githubusercontent.com/4047289/45295472-19f5cf00-b4ff-11e8-816b-acdbf45b0ee9.jpg)
